sdk/core/azure_core_test/src/proxy/models.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use azure_core::{json::to_json, RequestContent, StatusCode};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, fmt};
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Error {
+    #[serde(skip, default = "Error::default_status")]
+    pub status_code: StatusCode,
+    pub status: String,
+    pub message: String,
```

Comment:
The HTTP RFC is fairly emphatic that `status` should not be considered to be authoritative, and it may not be present at all. Does it make sense to consider `status` and `message` as `Option<String>`? 

Maybe another way of phrasing this is: Is it safe to treat an empty `status` and `message` as "not present"?

Reply:
Our service specs that defined this model say they are "required" which, in a response model, means they are always there. Given this is a client over a specific service, I'll take specificity over generalization.

Reply:
To put it another way, if the service does something it didn't spec correctly, I'd rather know about it and fix the service and/or client as problems arise so we're not incorrectly responding to undefined behavior.